lib/exercism/team_membership_invite.rb

```diff
@@ -0,0 +1,27 @@
+class TeamMembershipInvite < ActiveRecord::Base
+  belongs_to :team
+  belongs_to :user
+  belongs_to :inviter, class_name: 'User', foreign_key: :inviter_id
+
+  scope :for_team, -> (team_id) { where(team_id: team_id) }
+  scope :refused, -> { where(refused: true) }
+  scope :pending, -> { where(refused: false) }
+
+  def accept!
+    ActiveRecord::Base.transaction do
+      TeamMembership.create!(
+        team_id: team_id,
+        user_id: user_id,
+        inviter_id: inviter_id,
+        confirmed: true
+      )
+
+      destroy
+    end
+  end
+
+  def refuse!
+    self.refused = true
+    save
+  end
```

Comment:
So, when I was thinking in the membership request feature I imagined a scenario where every time the user gets rejected he requests a membership over and over, if we keep a refused membership we can create some business rule that allows a new requests only after one week, something like that. Then when I was coding this I realized that the inverse could also be true, a manager can invite an user every time he refuses to join.

I was thinking in implementing this after the `invite` and `request` features are done, but I don't know if I'm being overly cautious. ;)

Oh, we could also delete previously rejected memberships and keep only the last one.